core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -199,8 +204,8 @@ class AdminClient(val time: Time,
       metadata = describeConsumerGroupHandler(coordinator, groupId)
     }
 
-    if (!validateConsumerGroup(metadata))
-      throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType}' is not a valid consumer group")
+    if (timeoutMs > 0 && time.milliseconds - startTime > timeoutMs)
```

Comment:
Sorry to belabor this, but the order of the checks is a little weird. We may have timed out just as we have received a valid request (such as when the timeout is 0). Since we are ignoring all errors before the timeout has expired, couldn't we just do this?
```
if (!validateConsumerGroup(metadata))
  throw new TimeoutException("The consumer group command timed out while waiting for group to initialize")

// else return the result
```

One other nit: maybe `validateConsumerGroup` could be `isValidConsumerGroupResponse`?

Also, maybe we can log a DEBUG message if there are errors while retrying?